### PR TITLE
[build] Update Gazelle bazel dependency

### PR DESCRIPTION
**Pull request by drfloob**

### CODE REVIEW

**Comment by drfloob** [comment]
Fixes #36738

**Review by veblush — APPROVED** [review]
LGTM. Just make sure that it works with both Bazel 6.x and 7.x as we need to support Bazel 6.x for the next 6 months.

File: bazel/grpc_extra_deps.bzl [hunk 1]
```diff
@@ -52,7 +52,7 @@ def grpc_extra_deps(ignore_version_differences = False):
     api_dependencies()
 
     go_rules_dependencies()
-    go_register_toolchains(version = "1.18")
+    go_register_toolchains(version = "1.20")
```
Comment (yashykt) [inline bazel/grpc_extra_deps.bzl:55]: let's try using "host" here

**Comment by drfloob** [comment]
> LGTM. Just make sure that it works with both Bazel 6.x and 7.x as we need to support Bazel 6.x for the next 6 months.

Our tests would catch this, yes? If not, we need a bazel 6 test.

**Comment by veblush** [comment]
We do have Bazel 6 test so you'll know ;)

File: bazel/grpc_extra_deps.bzl (hunk 1 above)
Comment (drfloob) [reply bazel/grpc_extra_deps.bzl]: Done, let's see.

**Pull request closed without merging**